src/ophys_etl/workflows/on_prem/slurm/slurm.py

```diff
@@ -0,0 +1,399 @@
+import re
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from pathlib import Path
+from typing import Optional
+
+import logging
+
+import yaml
+from paramiko import SSHClient
+from pydantic import StrictInt, StrictStr, Field
+from simple_slurm import Slurm as SimpleSlurm
+
+from ophys_etl.workflows.app_config.app_config import app_config
+from ophys_etl.workflows.pipeline_module import PipelineModule
+from ophys_etl.workflows.utils.pydantic_model_utils import ImmutableBaseModel
+
+logger = logging.getLogger('airflow.task')
+
+
+class _SlurmSettings(ImmutableBaseModel):
+    cpus_per_task: StrictInt
+    mem: StrictInt
+    time: StrictStr = Field(description='Timeout, use format H:M:S')
+    gpus: Optional[StrictInt] = Field(
+        description='Number of GPUs',
+        default=0
+    )
+
+
+def _exec_slurm_command(command: str) -> str:
+    """Execute command over ssh on hpc-login"""
+    with SSHClient() as client:
+        client.load_system_host_keys()
+        client.connect('hpc-login')
+        _, stdout, stderr = client.exec_command(command=command)
+        stderr = stderr.read()
+        if stderr:
+            raise RuntimeError(stderr)
+        stdout = bytes.decode(stdout.read())
+    return stdout
+
+
+class SlurmState(Enum):
+    """States that a slurm job can be in"""
+    COMPLETED = 'COMPLETED'
```

Comment:
I took it from here: https://slurm.schedmd.com/squeue.html#SECTION_JOB-STATE-CODES
Actually even that list isn't exhaustive since when I cancelled a job it failed with CANCELLED_PLUS

Are you asking me to add this link to the docstring?

Reply:
I think that would be helpful, yes to give context to where the list comes from.